retrieval_service/datastore/providers/firestore.py

```diff
@@ -76,6 +79,68 @@ async def delete_collections(collection_list: list[AsyncCollectionReference]):
             [airports_ref, amenities_ref, flights_ref, policies_ref]
         )
 
+        async def delete_indexes(index_list):
+            # Check if the collection exists and deletes all indexes
+            delete_tasks = []
+            for index in index_list:
+                if index:
+                    delete_vector_index = [
+                        "gcloud",
+                        "alpha",
+                        "firestore",
+                        "indexes",
+                        "composite",
+                        "delete",
+                        index,
+                        "--database=(default)",
+                        "--quiet",  # Added to suppress delete warning
+                    ]
+
+                    delete_vector_index_process = await asyncio.create_subprocess_exec(
+                        *delete_vector_index,
+                    )
+                    delete_tasks.append(
+                        asyncio.create_task(delete_vector_index_process.wait())
+                    )
+            await asyncio.gather(*delete_tasks)
+
+        # List indexes and retrieve name field (file-path)
+        list_vector_index = [
+            "gcloud",
+            "alpha",
+            "firestore",
+            "indexes",
+            "composite",
+            "list",
+            "--database=(default)",
+            "--format=value(name)",  # prints name field
+        ]
+
+        list_vector_index_process = await asyncio.create_subprocess_exec(
+            *list_vector_index,
+            stdout=asyncio.subprocess.PIPE,
+        )
+
+        # Capture output and ignore stderr
+        stdout, __ = await list_vector_index_process.communicate()
+
+        # Decode and format output
+        indexes = stdout.decode().strip().split("\n")
+
+        # Check if the indexes already exist; if so, delete indexes
+        # Extract collection and index-id from file path
+        # Assign the index-id to the corresponding collection
+        if indexes == [""]:
+            pass
+        else:
+            collections = {"amenities": "", "policies": ""}
+            for line in indexes:
+                collection, index_id = line.split("/")[-3], line.split("/")[-1]
+                collections[collection] = index_id
+            amenities_ref = collections["amenities"]
+            policies_ref = collections["policies"]
+            await delete_indexes([amenities_ref, policies_ref])
```

Comment:
```suggestion
            collections = []
            for line in indexes:
                collection, index_id = line.split("/")[-3], line.split("/")[-1]
                collections.append(index_id)
            await delete_indexes(collections)
```

Reply:
I think explicitly stating which collection I'm passing and deleting might be more readable than just sending all the unidentified list of index-ids. Thoughts?

Reply:
I see. We can keep this then for the clarity. Do we need to change anything if "amenities" and "policies" doesn't exist or the name changes and do we not want to delete all indexes?

Reply:
That is checked inside `delete_indexes` function